tool/tsh/resolve_default_addr.go

```diff
@@ -0,0 +1,205 @@
+/*
+Copyright 2021 Gravitational, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"context"
+	"crypto/tls"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net"
+	"net/http"
+	"strconv"
+	"sync"
+	"time"
+
+	"github.com/gravitational/trace"
+	"github.com/sirupsen/logrus"
+)
+
+type raceResult struct {
+	addr string
+	err  error
+}
+
+// logResponseBody reads and dumps a response body to the log at the supplied
+// level. Note that it is still the caller's responsibility to close the body
+// stream.
+func logResponseBody(level logrus.Level, bodyStream io.Reader) {
+	if log.Logger.Level < level {
+		return
+	}
+
+	// NB: `ReadAll()` will time out (or be cancelled) according to the
+	//     context originally supplied to the request that initiated this
+	//     response, so no need to have an independent reading timeout
+	//     here.
+	body, err := ioutil.ReadAll(bodyStream)
```

Comment:
Should probably use `io.LimitReader` with some reasonable limit (what we expect legitimate ping endpoint to return) to be on a safe side.